### PR TITLE
Kafka producer/consumer

### PR DESCRIPTION
**Pull request by iancmcc**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,44 +1,46 @@
-ZENKIT_VERSION       := 1.3
+ZENKIT_VERSION       := 1.4
```
Comment (smousa) [inline Makefile]: do you want to move this into .env?

File: Makefile [hunk 2]
```diff
@@ -1,44 +1,46 @@
...
-							-v $(GOPATH)/src:/go/src:rw \
-							-e LOCAL_USER_ID=$(LOCAL_USER_ID) \
-							-e IN_DOCKER=1 \
-							-w /go/src/$(PACKAGE) \
-							$(BUILD_IMG)
-else
-DOCKER_CMD           :=
-endif
+DOCKER_PARAMS        := --rm -v /var/run/docker.sock:/var/run/docker.sock \
+						     -v $(ROOTDIR):/go/src/$(PACKAGE):rw \
```
Comment (smousa) [inline Makefile]: the spacing is weird here

File: databus/consumer.go [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	// Get errors and notifications first
+	stop := false
+	for !stop {
+		select {
+		case <-c.con.Errors():
+		// TODO
```
Comment (smousa) [inline databus/consumer.go]: TODO?

Comment (smousa) [inline databus/consumer.go]: There are others, I am not going to spam this review with them

File: databus/consumer.go [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+}
+
+func (c *saramaClusterDatabusConsumer) validateType(message interface{}) (int, int, error) {
+	// Make sure our message type is valid
+	//  Must be a pointer to struct with fields tagged as follows:
+	//   `zenkit:"message-key"`
+	//   `zenkit:"message-value"`
+	messageType := reflect.TypeOf(message)
+	if messageType.Kind() != reflect.Ptr {
+		return 0, 0, errors.Wrap(ErrInvalidMessageType, "type is not a pointer")
```
Comment (smousa) [inline databus/consumer.go]: Make sure you document these errors

File: databus/producer_test.go [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+	"fmt"
+	"math/rand"
+	"strconv"
+
+	"github.com/datamountaineer/schema-registry"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+type mockSyncProducer struct {
```
Comment (smousa) [inline databus/producer_test.go]: Why not use mockery to build this?

**Comment by smousa** [comment]
The code may be too abstracted for me to fully understand.  Please add more comments.

File: Makefile (hunk 1 above)
Comment (iancmcc) [inline Makefile]: No, there's no docker-compose stuff here. That would be an unnecessary abstraction, I think.

File: databus/consumer.go (hunk 3 above)
Comment (iancmcc) [inline databus/consumer.go]: They're on purpose. They're not worth implementing at this stage, but we need to bleed off the channel, and we will need them eventually.

File: databus/consumer.go (hunk 4 above)
Comment (iancmcc) [inline databus/consumer.go]: Where? These are just some we wrapped.

File: databus/producer_test.go (hunk 5 above)
Comment (iancmcc) [inline databus/producer_test.go]: Easier and clearer to do it here, and we don't need a full mock.

**Comment by iancmcc** [comment]
retest this

**Pull request merged**